src/management-system-v2/app/(dashboard)/[environmentId]/layout-client.tsx

```diff
@@ -58,6 +65,11 @@ const Layout: FC<
     (item) => !(breakpoint.xs && item && 'type' in item && item.type === 'divider'),
   );
 
+  const setUserSpaces = useUserSpaces((state) => state.setUserSpaces);
+  useEffect(() => {
+    setUserSpaces(userEnvironments);
+  });
+
```

Comment:
This effect is not recommended for initial state, see these approaches instead: https://docs.pmnd.rs/zustand/guides/initialize-state-with-props

Reply:
I simply changed the zustand store for a context.